web_monitoring/diffing_server.py

```diff
@@ -48,8 +60,23 @@ def get(self, differ):
         a = query_params.pop('a')
         b = query_params.pop('b')
 
-        # Fetch server response for URLs a and b.
-        res_a, res_b = yield [client.fetch(a), client.fetch(b)]
+        # Special case for local files, for dev/testing.
+        if a.startswith('file://') and b.startswith('file://'):
+            if os.get('WEB_MONITORING_APP_ENV') in ('development', 'testing'):
+                headers = {'Content-Type': 'application/html; charset=UTF-8'}
+                with open(a[7:], 'rb') as f:
+                    body = f.read()
+                    res_a = MockResponse(a, body, headers)
+                with open(b[7:], 'rb') as f:
+                    body = f.read()
+                    res_b = MockResponse(b, body, headers)
+            else:
+                self.send_error(
+                    500, reason=("Local files can only be used in development "
+                                 "or testing environment."))
+        else:
```

Comment:
What happens when only one of the arguments is a local file? Although I don't see that happening frequently, I think the code should cover all cases.

Reply:
That's a valid point. Rewriting to deal with that is probably a ~30-line change, and we're planning to delete all this code (in favor of a queue-based system) in ~weeks, so I think I'll punt if unless anyone feels strongly about it.

Reply:
I think we can skip it if the queue-based system isn't far.

Reply:
> and we're planning to delete all this code (in favor of a queue-based system) in ~weeks

Wait, are we? I thought queues were only meant to facilitate knowing when new versions have been added in order to do pre-analysis. We would still use requests for generalized diffing (e.g. when someone wants to compare version 1 and version 5 of a page), wouldn’t we?